command/genesis/params.go

```diff
@@ -319,3 +369,69 @@ func (p *genesisParams) getResult() command.CommandResult {
 		Message: fmt.Sprintf("Genesis written to %s\n", p.genesisPath),
 	}
 }
+
+func parseECDSAValidators(set validators.ValidatorSet, values []string) (validators.ValidatorSet, error) {
+	addrs := make([]types.Address, 0)
+
+	if set != nil {
+		valSet, ok := set.(*validators.ECDSAValidatorSet)
+		if !ok {
+			return nil, fmt.Errorf("invalid validator set, expected ECDSAValidatorSet but got %T", set)
+		}
+
+		addrs = []types.Address(*valSet)
+	}
+
+	for _, v := range values {
+		bytes, err := hex.DecodeString(strings.TrimPrefix(v, "0x"))
+		if err != nil {
+			return nil, err
+		}
+
+		addrs = append(addrs, types.BytesToAddress(bytes))
+	}
+
+	newValSet := validators.ECDSAValidatorSet(addrs)
+
+	return &newValSet, nil
+}
+
+func parseBLSValidators(set validators.ValidatorSet, values []string) (validators.ValidatorSet, error) {
+	vals := make([]validators.BLSValidator, 0)
+
+	if set != nil {
+		valSet, ok := set.(*validators.BLSValidatorSet)
+		if !ok {
+			return nil, fmt.Errorf("invalid validator set, expected BLSValidatorSet but got %T", set)
+		}
+
+		vals = []validators.BLSValidator(*valSet)
+	}
+
+	for _, value := range values {
+		subValues := strings.Split(value, ":")
+
+		if len(subValues) != 2 {
+			return nil, fmt.Errorf("invalid validator format, expected [Validator Address]:[BLS Public Key]")
+		}
+
+		addrBytes, err := hex.DecodeString(strings.TrimPrefix(subValues[0], "0x"))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse address: %w", err)
+		}
+
+		pubKeyBytes, err := hex.DecodeString(strings.TrimPrefix(subValues[1], "0x"))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse BLS Public Key: %w", err)
+		}
+
+		vals = append(vals, validators.BLSValidator{
+			Address:   types.BytesToAddress(addrBytes),
+			BLSPubKey: pubKeyBytes,
+		})
+	}
+
+	newValSet := validators.BLSValidatorSet(vals)
+
+	return &newValSet, nil
+}
```

Comment:
What do you think about moving these methods out do some better package, like `validators`?

This is not logic that is specific for the genesis params

Reply:
Good idea. Moved at b8b41249444ab5cda3c156f251c6a9ced84bb5f1